packages/shared/test/shared-secret-compatibility.test.ts

```diff
@@ -0,0 +1,271 @@
+        : null,
+      rustGeneratedCiphertext: vector.rust_generated_ciphertext
+        ? new Uint8Array(vector.rust_generated_ciphertext)
+        : null,
+      expectedPlaintext: vector.expected_plaintext,
+    };
+  };
+
+  // Parse test vectors into usable formats
+  const testVector1 = prepareTestVector(test_vectors[0]);
+  const testVector2 = prepareTestVector(test_vectors[1]);
+  const rustCompatVector = prepareTestVector(test_vectors[2]);
+
+  // Set up a fixed nonce for testing
+  function setupFixedNonceMock(fixedNonce: Uint8Array) {
+    // Get the mocked function
+    const mockedRandomBytes = vi.mocked(randomBytes);
+
+    // Clear previous implementations
+    mockedRandomBytes.mockReset();
+
+    // Configure it to return our fixed nonce
+    mockedRandomBytes.mockImplementation((bytesLength?: number) => {
+      // If length is undefined or matches our nonce length, return the fixed nonce
+      // Otherwise return a new array of the requested length
+      if (bytesLength === undefined || bytesLength === fixedNonce.length) {
+        return fixedNonce;
+      } else {
+        return new Uint8Array(bytesLength);
+      }
+    });
+
+    // Return a cleanup function
+    return () => {
+      mockedRandomBytes.mockReset();
+    };
+  }
+
+  describe('Round-trip fixed vector tests', () => {
+    // Testing each vector for round-trip compatibility
+    it('should correctly encrypt and decrypt test vector 1', () => {
+      const { sharedSecret, plaintext, fixedNonce, expectedCiphertext } =
+        testVector1;
+
+      // Set up our mock to return the fixed nonce
+      const cleanupMock = setupFixedNonceMock(fixedNonce);
+
+      try {
+        // Encrypt with fixed nonce
+        const ciphertext = encryptWithSharedSecret(sharedSecret, plaintext);
+
+        // Check against expected ciphertext from the JSON file
+        const ciphertextHex = Buffer.from(ciphertext).toString('hex');
+        if (expectedCiphertext) {
+          const expectedHex = Buffer.from(expectedCiphertext).toString('hex');
+          expect(ciphertextHex).toEqual(expectedHex);
+          console.debug(
+            `✓ Vector ${testVector1.id} ciphertext matches expected output`,
+          );
+        } else {
+          console.error(
+            `Generated ciphertext for vector ${testVector1.id}:`,
+            ciphertextHex,
+          );
+          console.error(
+            'Add this to the test vectors JSON file to enable validation',
+          );
+        }
+
+        // Verify decryption works correctly
+        const decrypted = decryptWithSharedSecret(sharedSecret, ciphertext);
+        expect(Buffer.from(decrypted).toString()).toEqual(
+          Buffer.from(plaintext).toString(),
+        );
+      } finally {
+        // Always restore the original implementation
+        cleanupMock();
+      }
+    });
+
+    it('should correctly encrypt and decrypt test vector 2 (empty plaintext)', () => {
```

Comment:
Given that we have test vectors, how come we need separate `it` blocks for each vector? Are all the test vectors not structured the same? If not, can we set it up to not need repeated code - can `it.each` be leveraged?

Reply:
You are very right. There is no real need for separate `it` blocks. I unified the vectors fileds keeping just the minimal differences. And used `it.each`. However, I kept the code checking the errors as is.